event/event.go

```diff
@@ -0,0 +1,208 @@
+// the execution of a program.
+package event
+
+// Event is an interface that custom events should implement.
+// It is empty for now because there are no general methods
+// required of events yet.
+type Event interface {
+}
+
+// KeyCharacter is an event that occurs when a character is actually typed on
+// the keyboard. This may be provided by an input method.
+type KeyCharacter struct {
+	// Code is the key code of the key typed.
+	// TODO: this should change later from an int to an enumeration type.
+	Code int
+	// Modifiers are the modifiers pressed together with the key.
+	// TODO: this should change later from an int to an enumeration type.
+	Modifiers int
+	// Character is the character that was typed.
+	Character rune
+}
+
+// KeyDown is an event that occurs when a key is pressed on the keyboard.
+type KeyDown struct {
+	// Code is the key code of the key pressed or released.
+	// TODO: this should change later from an int to an enumeration type.
+	Code int
+	// Modifiers are the modifiers pressed together with the key.
+	// TODO: this should change later from an int to an enumeration type.
+	Modifiers int
+}
+
+// KeyUp is an event that occurs when a key is released on the keyboard.
+// The data is the same as for a KeyDown event.
+type KeyUp KeyDown
+
+// GamepadAxis is for event where an axis on a gamepad changes.
+type GamepadAxis struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Axis is the axis of the game pad that changed position.
+	Axis int
+	// Position is the psoition of the axis after the change.
+	// It varies between -1.0 and 1.0.
+	Position float32
+}
+
+// GamepadButtonDown is a gamepad button press event.
+type GamepadButtonDown struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Button is the button that was pressed on the game pad.
+	Button int
+	// Pressure is the pressure that is applied to the gamepad button.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// GamepadButtonDown is a gamepad button release event.
+// The data is identical to a GamePadButtonDown event.
+type GamepadButtonUp GamepadButtonDown
+
+// GamepadAttach happens when a new gamepad is attached.
+type GamepadAttach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Axes represents the amount of axes the gamepad has.
+	Axes int
+	// Buttons represents the amount of buttons the gamepad has.
+	Buttons int
+}
+
+// GamepadDetach happens when a gamepad is detached.
+type GamepadDetach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+}
+
+// MouseMove is a mouse movement event.
+type MouseMove struct {
+	// X is the X position of the mouse pointer.
+	X float32
+	// Y is the Y position of the mouse pointer.
+	Y float32
+	// DeltaX is the change in X since the last MouseMove event.
+	DeltaX float32
+	// DeltaY is the change in Y since the last MouseMove event.
+	DeltaY float32
```

Comment:
* Use int? Or some devices usually use float32 precision?
* Let's add explanation that the unit is device-independent pixels.

Reply:
Let's use float32 expressed as device independent pixels, for the reasons mentioned above.